cmd/dvol-docker-plugin/dvol-docker-plugin.go

```diff
@@ -0,0 +1,167 @@
+package main
+
+import (
+    "fmt"
+    "encoding/json"
+    "errors"
+    "log"
+    "net"
+    "net/http"
+    "os"
+    "io/ioutil"
+
+	"github.com/ClusterHQ/dvol/pkg/api"
+)
+
+const PLUGINS_DIR = "/run/docker/plugins"
+const DVOL_SOCKET = PLUGINS_DIR + "/dvol.sock"
+const VOL_DIR = "/var/lib/dvol/volumes"
+
+type ResponseImplements struct {
+    // A response to the Plugin.Activate request
+    Implements []string
+}
+
+type RequestCreate struct {
+    // A request to create a volume for Docker
+    Name string
```

Comment:
I tried this, but it would have resulted in this:
```
cmd/dvol-docker-plugin/dvol-docker-plugin.go:80: cannot use []string literal (type []string) as type []DockerVolumePluginType in field value
cmd/dvol-docker-plugin/dvol-docker-plugin.go:95: cannot use name (type VolumeName) as type string in argument to dvol.VolumeExists
cmd/dvol-docker-plugin/dvol-docker-plugin.go:98: cannot use name (type VolumeName) as type string in argument to dvol.CreateVolume
cmd/dvol-docker-plugin/dvol-docker-plugin.go:126: cannot use name (type VolumeName) as type string in argument to dvol.VolumeExists
cmd/dvol-docker-plugin/dvol-docker-plugin.go:127: cannot use name (type VolumeName) as type string in argument to dvol.SwitchVolume
cmd/dvol-docker-plugin/dvol-docker-plugin.go:172: cannot use errString (type string) as type ErrResponse in field value
```
i didn't want to pepper these type changes throughout the code so i reverted the change